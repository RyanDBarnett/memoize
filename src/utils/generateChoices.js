import React from 'react';

function generateChoices(choices, showNextQuestion, checkAnswer) {
  const handleClick = (e) => {
    let answer = e.target.innerHTML;
    checkAnswer(answer);
    showNextQuestion();
  }
  return (
    <div className="btnContainer">
      {
        choices.map((choice, index) => {
          return <button aria-label={"Answer option: " + choice} className={'choice-btn-' + (index + 1)} key={index} onClick={handleClick}>{choice}</button>
        })
      }
    </div>
  )
}

export default generateChoices;