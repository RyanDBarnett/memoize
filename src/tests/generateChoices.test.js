import generateChoices from '../utils/generateChoices';
import { shallow } from "enzyme";

describe('generateChoices', () => {
  let wrapper;

  const mockChoices = ['Yes','No'];
  const mockshowNextQuestion = jest.fn();
  const mockcheckAnswer = jest.fn();

  beforeEach( () => {
    wrapper = shallow(
      generateChoices(mockChoices, mockshowNextQuestion, mockcheckAnswer)
    )
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke check answer ', () => {
    let choiceBtn = wrapper.find('.choice-btn-1');
    choiceBtn.simulate('click', {target: {innerHTML: 'Test'}});
    expect(mockcheckAnswer).toHaveBeenCalled();
    expect(mockshowNextQuestion).toHaveBeenCalled();
  });
});