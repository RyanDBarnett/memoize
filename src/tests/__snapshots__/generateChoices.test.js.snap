// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateChoices should match the snapshot 1`] = `
<div
  className="btnContainer"
>
  <button
    className="choice-btn-1"
    key="0"
    onClick={[Function]}
  >
    Yes
  </button>
  <button
    className="choice-btn-2"
    key="1"
    onClick={[Function]}
  >
    No
  </button>
</div>
`;
